src/universal/modules/meeting/ducks/meeting.js

```diff
@@ -0,0 +1,166 @@
+export const UPDATE_MEETING_ERROR = 'action/meeting/UPDATE_MEETING_ERROR';
+export const UPDATE_MEETING_SUCCESS = 'action/meeting/UPDATE_MEETING_SUCCESS';
+//TODO multiple meetings at once? It's possible with redux-operations
+// making the switch to redux-operations now is cheap, later on it'll become a pain to switch
+
+const initialState = iMap({
+  isLoading: false,
+  isLoaded: false,
+  instance: iMap({
+    id: '',
+    content: '',
+    lastUpdatedBy: '',
+    currentEditors: iList()
+  }),
+  mySocketId: '',
+  otherEditing: false
+});
+
+export default function reducer(state = initialState, action = {}) {
+  switch (action.type) {
+    case SET_MEETING_ID:
+      return state.mergeDeep({
+        instance: iMap({
+          id: action.payload.id
+        })
+      })
+    case CREATE_MEETING_REQUEST:
+      return state.merge({
+        isLoading: true
+      });
+    case CREATE_MEETING_SUCCESS:
+      return state.mergeDeep({
+        isLoading: false,
+        isLoaded: true,
+        instance: iMap({
+          id: action.payload.id
+        })
+      });
+    case CREATE_MEETING_ERROR:
+    case UPDATE_MEETING_ERROR:
+      return state.merge({
+        isLoading: false,
+        error: action.error
+      });
+    case UPDATE_MEETING_SUCCESS:
+      return state.mergeDeep({
+        instance: iMap(action.payload)
+      });
+    default:
+      return state
+  }
+};
+
+export const setMeetingId = meetingId => {
+  return {
+    type: SET_MEETING_ID,
+    payload: {id: meetingId}
+  }
+}
+
+export const createMeetingAndRedirect = () => {
+  return async function (dispatch, getState) {
+    const query = `
+    mutation {
+      payload: createMeeting {
+        id
+      }
+    }`;
+    const {error, data} = await fetchGraphQL({query});
+    if (error) {
+      return dispatch({type: CREATE_MEETING_ERROR, error});
+    }
+    const {payload} = data;
+    dispatch({type: CREATE_MEETING_SUCCESS, payload});
+    const id = ensureState(getState()).getIn(['meeting', 'instance', 'id']);
+    // replace, don't use push. a click should go back 2.
+    dispatch(replace(`/meeting/${id}`));
+  };
+}
+
+const updateMeetingSuccess = (payload, meta) => {
+  return {
+    type: UPDATE_MEETING_SUCCESS,
+    payload,
+    meta
+  }
+};
+
+export const loadMeeting = meetingId => {
+  const sub = `getMeeting/${meetingId}`;
+  const {authTokenName} = localStorageVars;
+  const socket = socketCluster.connect({authTokenName});
+  socket.subscribe(sub, {waitForAuth: true});
+  return dispatch => {
+    // client-side changefeed handler
+    socket.on(sub, data => {
+      console.log('DATA', data)
+      dispatch({
+        type: UPDATE_MEETING_SUCCESS,
+        payload: data,
+        meta: {synced: true}
+      })
+    });
+    socket.on('unsubscribe', channelName => {
+      if (channelName === sub) {
+        console.log(`unsubbed from ${channelName}`)
+      }
+    });
+  };
+}
+
+export const updateEditing = (meetingId, editor, isEditing) => {
```

Comment:
I noticed in your video demo and in my testing that there is a tendency to drop characters here. [The way I 'solved' for this was a bit hacky](https://github.com/ParabolInc/action/blob/master/src/redux/modules/meeting.js#L174): I'd optimistically update the local state while using underscore's throttle method to batch updates (a sort of poor-man's NAGLE algorithm). Curious in the future what strategies we develop together to avoid punishing the backend! :smile: